demo/runtime/src/runtime/council.rs

```diff
@@ -0,0 +1,1392 @@
+	}
+}
+
+/// Finalise the vote, removing each of the `removals` and inserting `seats` of the most approved
+/// candidates in their place. If the total council members is less than the desired membership
+/// a new vote is started.
+/// Clears all presented candidates, returning the bond of the elected ones.
+fn finalise_tally() {
+	storage::kill(SNAPSHOTED_STAKES);
+	let (_, coming, expiring): (BlockNumber, u32, Vec<AccountId>) = storage::take(NEXT_FINALISE)
+		.expect("finalise can only be called after a tally is started.");
+	let leaderboard: Vec<(Balance, AccountId)> = storage::take(LEADERBOARD).unwrap_or_default();
+	let new_expiry = system::block_number() + term_duration();
+
+	// return bond to winners.
+	let candidacy_bond = candidacy_bond();
+	for &(_, ref w) in leaderboard.iter()
+		.rev()
+		.take_while(|&&(b, _)| b != 0)
+		.take(coming as usize)
+	{
+		staking::internal::set_balance(w, staking::balance(w) + candidacy_bond);
+	}
+
+	// set the new council.
+	let new_council: Vec<_> = active_council()
+		.into_iter()
+		.skip(expiring.len())
+		.chain(leaderboard.iter()
+			.rev()
+			.take_while(|&&(b, _)| b != 0)
+			.take(coming as usize)
+			.cloned()
+			.map(|(_, a)| (a, new_expiry)))
+		.collect();
+	storage::put(ACTIVE_COUNCIL, &new_council);
+
+	// clear all except runners-up from candidate list.
+	let candidates: Vec<AccountId> = storage::get_or_default(CANDIDATES);
+	let mut new_candidates = vec![AccountId::default(); candidates.len()];	// shrink later.
+	let runners_up = leaderboard.into_iter()
+		.rev()
+		.take_while(|&(b, _)| b != 0)
+		.skip(coming as usize)
+		.map(|(_, a)| (a, candidate_reg_info(&a).expect("runner up must b registered").1));
+	let mut count = 0u32;
+	for (address, slot) in runners_up {
+		new_candidates[slot as usize] = address;
+		count += 1;
+	}
+	for (old, new) in candidates.iter().zip(new_candidates.iter()) {
+		if old != new {
+			// removed - kill it
+			storage::kill(&old.to_keyed_vec(REGISTER_INFO_OF));
+		}
+	}
+	// discard any superfluous slots.
+	if let Some(last_index) = new_candidates.iter().rposition(|c| *c != AccountId::default()) {
+		new_candidates.truncate(last_index + 1);
+	}
+	storage::put(CANDIDATES, &(new_candidates));
+	storage::put(CANDIDATE_COUNT, &count);
+	storage::put(VOTE_COUNT, &(vote_index() + 1));
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use runtime_io::{with_externalities, twox_128, TestExternalities};
+	use codec::{KeyedVec, Joiner};
+	use keyring::Keyring;
+	use environment::with_env;
+	use demo_primitives::{AccountId, Proposal};
+	use runtime::{staking, session, democracy};
+
+	fn new_test_ext() -> TestExternalities {
+		let alice = Keyring::Alice.to_raw_public();
+		let bob = Keyring::Bob.to_raw_public();
+		let charlie = Keyring::Charlie.to_raw_public();
+		let dave = Keyring::Dave.to_raw_public();
+		let eve = Keyring::Eve.to_raw_public();
+		let ferdie = Keyring::Ferdie.to_raw_public();
+		let one = Keyring::One.to_raw_public();
+
+		map![
```

Comment:
should use constants for this rather than arcane string constants.

Reply:
should add a helper in `genesismap` anyway

Reply:
i guess we can export the constants.

Reply:
ideally would do something like #78

Reply:
yup.